schemes/rayleigh_friction/rayleigh_friction.F90

```diff
@@ -0,0 +1,160 @@
+    raykrange = raykrange_nl
+    rayk0 = rayk0_nl
+
+    if ( raytau0 > 0._kind_phys) then  ! Rayleigh Friction enabled
+       ! Allocate module data
+       allocate(otau(pver), stat=ierr)
+       if (ierr /= 0) then
+          errflg = ierr
+          errmsg = trim(subname)//': Allocate of otau failed'
+          return
+       end if
+
+       !-----------------------------------------------------------------------
+       ! Compute tau array
+       !-----------------------------------------------------------------------
+
+       krange = raykrange
+       if (raykrange .eq. 0._kind_phys) krange = (rayk0 - 1) / 2._kind_phys
+
+       tau0 = (86400._kind_phys) * raytau0   ! convert to seconds
+       otau0 = 0._kind_phys
+       if (tau0 .ne. 0._kind_phys) otau0 = 1._kind_phys/tau0
+
+       do k = 1, pver
+          x = (rayk0 - k) / krange
+          otau(k) = otau0 * (1 + tanh(x)) / (2._kind_phys)
+       enddo
+    end if ! Rayleigh Friction enabled
+
+    if ( masterproc ) then
+       if (raytau0 > 0._kind_phys) then
+          write(iulog,*) 'tuning parameters rayleigh_friction_init: rayk0',rayk0
+          write(iulog,*) 'tuning parameters rayleigh_friction_init: raykrange',raykrange
+          write(iulog,*) 'tuning parameters rayleigh_friction_init: raytau0',raytau0
+       else
+          write(iulog,*) 'Rayleigh Friction not enabled.'
+       endif
+    endif
+
+  end subroutine rayleigh_friction_init
+
+  !=========================================================================================
+  !> \section arg_table_rayleigh_friction_run Argument Table
+  !! \htmlinclude rayleigh_friction_run.html
+  subroutine rayleigh_friction_run(ncol, pver, ztodt, u, v, dudt, dvdt, dsdt, errmsg, errflg)
+
+    !------------------------------Arguments--------------------------------
+    integer,             intent(in) :: ncol  
+    integer,             intent(in) :: pver
+    real(kind_phys),     intent(in) :: ztodt   !physics timestep
+    real(kind_phys),     intent(in) :: u(:,:)     
+    real(kind_phys),     intent(in) :: v(:,:)     
+    real(kind_phys),    intent(out) :: dudt(:,:) !tendency_of_eastward_wind    
+    real(kind_phys),    intent(out) :: dvdt(:,:) !tendency_of_northward_wind    
+    real(kind_phys),    intent(out) :: dsdt(:,:)  !heating_rate 
+      
+    character(len=512), intent(out) :: errmsg
+    integer,            intent(out) :: errflg
+   
+    !---------------------------Local storage-------------------------------
+    character(len=*),     parameter :: subname = 'rayleigh_friction_run'
+    integer :: k                                   ! level
+    real(kind_phys) :: rztodt                      ! 1./ztodt
+    real(kind_phys) :: c1, c2, c3                  ! temporary variables
+    !-----------------------------------------------------------------------
+    
+    ! initialize values
+    errmsg = ''
+    errflg = 0
+    dudt(:,:)=0._kind_phys
+    dvdt(:,:)=0._kind_phys
+    dsdt(:,:) =0._kind_phys
```

Comment:
These initialization lines can be deleted as lower down will set all of the values.  The corresponding ESCOMP/CAM PR will do this initialization instead and ESCOMP/CAM-SIMA does not need it.

Reply:
Done